qa/L0_state_management/check_results.py

```diff
@@ -85,8 +85,8 @@ def check_interrupt_handling(self):
         if log_contents.find(token) == -1:
             return False
 
-        # check that 2nd model is profiled once
-        token = f"Profiling {self._profile_models[1]}"
+        # check that 1st model is profiled twice
```

Comment:
I don't understand this change/comment. In the test.sh it states that it is going to throw an interrupt when it detects the creation of the `1.ckpt` file.
With your change my understanding is that we would not be creating such a file (the 0.ckpt would contain all 3 models). Can you confirm what the new behavior is now?

Reply:
In the `test.sh` script, at line 109, script will wait until the first checkpoint is created.

Previously, the MA would create a checkpoint after completing the model profiling. So when the `0.ckpt` checkpoint was created(indicating completion of profiling two configurations for the first model), the script would break the loop and generate an interrupt. Also meanwhile, the first configuration of the second model would also get profiled.

Now, with the updated approach, a checkpoint is written for every configuration (multiple times for the same model). So MA will generate a checkpoint after the first configuration of the first model is executed. As a result, script would break the loop and only the second configuration of the first model will be profiled thereafter, and the first configuration of the second model will never be started.

So I modified the logic to check only for the first model, to make sure 2 configurations are profiled.

Reply:
Is this working consistently for you? I'm worried it's possible to potentially sneak a second profile within the 5 second sleep + kill. 

Please change the comment in `test.sh` to indicate that a single configuration of the first model will be profiled before the interrupt occurs